crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -621,16 +621,54 @@ fn response_to_capture_update(
     response_list: HashMap<String, CaptureSyncResponse>,
 ) -> RouterResult<Vec<(storage::Capture, storage::CaptureUpdate)>> {
     let mut capture_update_list = vec![];
+    let mut unmapped_captures = vec![];
     for (connector_capture_id, capture_sync_response) in response_list {
         let capture =
             multiple_capture_data.get_capture_by_connector_capture_id(connector_capture_id);
         if let Some(capture) = capture {
             capture_update_list.push((capture.clone(), capture_sync_response.try_into()?))
+        } else {
+            // connector_capture_id may not be populated in the captures table in some case
+            // if so, we try to map the unmapped capture response and captures in DB.
+            unmapped_captures.push(capture_sync_response)
         }
     }
+    capture_update_list.extend(get_capture_update_for_unmapped_capture_responses(
+        unmapped_captures,
+        multiple_capture_data,
+    )?);
+
     Ok(capture_update_list)
 }
 
+fn get_capture_update_for_unmapped_capture_responses(
+    unmapped_capture_sync_response_list: Vec<CaptureSyncResponse>,
+    multiple_capture_data: &MultipleCaptureData,
+) -> RouterResult<Vec<(storage::Capture, storage::CaptureUpdate)>> {
+    let mut result = Vec::new();
+    let captures_without_connector_capture_id: Vec<_> = multiple_capture_data
+        .get_pending_captures_without_connector_capture_id()
+        .into_iter()
+        .cloned()
+        .collect();
+    for capture_sync_response in unmapped_capture_sync_response_list {
+        if let Some(capture) = captures_without_connector_capture_id
+            .iter()
+            .find(|capture| {
+                capture_sync_response.get_connector_response_reference_id()
+                    == Some(capture.capture_id.clone())
+                    || capture_sync_response.get_amount_captured() == Some(capture.amount)
```

Comment:
what if two captures have the same amount?

Reply:
Then we map such captures in any order, wouldn't be an issue since, amount will be same.